### PR TITLE
#52 update candidate if email exists

### PR DESCRIPTION
**Pull request by alesyalis**

### CODE REVIEW

File: Recrutify/Recrutify.Host/Configuration/Profiles/CandidateProfile.cs [hunk 1]
```diff
@@ -13,7 +12,7 @@ public CandidateProfile()
             CreateMap<CandidateCreateDTO, Candidate>()
                 .ForMember(x => x.ProjectResults, opt => opt.Ignore())
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Host/Configuration/Profiles/CandidateProfile.cs:14]: rename all x to dest, cause somewhere is dest somewhere is x

File: Recrutify/Recrutify.Services/Services/CandidateService.cs [hunk 2]
```diff
@@ -50,12 +50,27 @@ public IQueryable<CandidateDTO> Get()
...
+            {
+                await _candidateRepository.CreateAsync(candidate);
+                var newCanditate = _mapper.Map<CandidateDTO>(candidate);
+                newCanditate.Id = Guid.NewGuid();
+                return newCanditate;
+            }
+
+            var candidateToUpdate = _mapper.Map(candidateCreateDTO, currentCandidate.DeepCopy());
+            var primarySkill = _mapper.Map<CandidatePrimarySkill>(candidateCreateDTO.PrimarySkill);
+            var newProjectResult = new ProjectResult { ProjectId = projectId, PrimarySkill = primarySkill };
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Services/Services/CandidateService.cs:67]: put it to 68 line, you should declare vars only when you need it, close to first using

File: recrutify/Recrutify.Host/Configuration/Profiles/CandidateProfile.cs [hunk 3]
```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using Recrutify.DataAccess.Models;
+using Recrutify.Services.DTOs;
+
+namespace Recrutify.Host.Configuration.Profiles
```
Comment (elizavetapavliv) [inline recrutify/Recrutify.Host/Configuration/Profiles/CandidateProfile.cs:5]: It's again file in recrutify folder

File: Recrutify/Recrutify.Services/Services/CandidateService.cs (hunk 2 above)
Comment (alesyalis) [reply Recrutify/Recrutify.Services/Services/CandidateService.cs:68]: done

File: Recrutify/Recrutify.Host/Configuration/Profiles/CandidateProfile.cs (hunk 1 above)
Comment (alesyalis) [reply Recrutify/Recrutify.Host/Configuration/Profiles/CandidateProfile.cs:14]: done

File: Recrutify/Recrutify.Services/Services/CandidateService.cs [hunk 4]
```diff
@@ -50,12 +50,27 @@ public IQueryable<CandidateDTO> Get()
...
+            {
+                candidate.Id = Guid.NewGuid();
+                await _candidateRepository.CreateAsync(candidate);
+                var newCanditate = _mapper.Map<CandidateDTO>(candidate);
+                return newCanditate;
+            }
+
+            var candidateToUpdate = _mapper.Map(candidateCreateDTO, currentCandidate.DeepCopy());
+            var primarySkill = _mapper.Map<CandidatePrimarySkill>(candidateCreateDTO.PrimarySkill);
+            var projectResults = currentCandidate?.ProjectResults?.ToList();
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Services/Services/CandidateService.cs:67]: currentCandidate cannot be null, you have checked it with if before, remove ? after it

File: Recrutify/Recrutify.Services/Services/CandidateService.cs [hunk 5]
```diff
@@ -50,12 +50,27 @@ public IQueryable<CandidateDTO> Get()
...
+            var newProjectResult = new ProjectResult { ProjectId = projectId, PrimarySkill = primarySkill };
+            projectResults?.Add(newProjectResult);
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Services/Services/CandidateService.cs:69]: if projectResults are null, check it and init it with new list, with one element newProjectResults, if not null just add new element to it, like you wrote now

Comment (alesyalis) [reply Recrutify/Recrutify.Services/Services/CandidateService.cs]: done

File: Recrutify/Recrutify.Services/Services/CandidateService.cs (hunk 4 above)
Comment (alesyalis) [reply Recrutify/Recrutify.Services/Services/CandidateService.cs:67]: done

File: recrutify/Recrutify.Host/Configuration/Profiles/CandidateProfile.cs (hunk 3 above)
Comment (alesyalis) [reply recrutify/Recrutify.Host/Configuration/Profiles/CandidateProfile.cs]: done

File: Recrutify/Recrutify.Services/Services/CandidateService.cs [hunk 6]
```diff
@@ -50,12 +50,32 @@ public IQueryable<CandidateDTO> Get()
...
+            var candidateToUpdate = _mapper.Map(candidateCreateDTO, currentCandidate.DeepCopy());
+            var primarySkill = _mapper.Map<CandidatePrimarySkill>(candidateCreateDTO.PrimarySkill);
+            var projectResults = currentCandidate?.ProjectResults?.ToList();
+            var newProjectResult = new ProjectResult { ProjectId = projectId, PrimarySkill = primarySkill };
+            if (projectResults == null)
+            {
+                projectResults = new List<ProjectResult> { newProjectResult };
+            }
+
+            projectResults.Add(newProjectResult);
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Services/Services/CandidateService.cs:74]: Write it in else

Comment (alesyalis) [reply Recrutify/Recrutify.Services/Services/CandidateService.cs]: done

**Review by elizavetapavliv — APPROVED** [review]

**Pull request merged**